### PR TITLE
Add tests for external dependencies in GCP

### PR DESCRIPTION
**Pull request by nandajavarma**
## Description

This PR adds the external dependencies test for GCP cloud platform. These dependencies include `cloudsql` as external database, `GCR` as external registry, and `cloud storage` as external storage. These dependencies are randomly selected for each test providing us a overall testing of every combination over a course of days. 

## Related Issue(s)

Fixes #

## How to test

You can run the following command to test one of the available combinations:
```
werft run github -j .werft/gke-installer-tests.yaml
```

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@nandajavarma: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: .werft/installer-tests.ts
Comment (adrienthebo) [inline .werft/installer-tests.ts:121]: Suggestion: performing randomization in this location means that loading this file isn't an idempotent operation. If a randomized test configuration fails we'll likely need to repeat that test case (either to rule out a transient failure or reproduce the failure). Because these settings are encoded during the file evaluation we don't have a means of injecting an exact configuration.

Phase parameterization can be kept out of scope for this PR (espcially because it'll take time to define how we configure phases), but we should definitely follow up on this in another issue/PR for Q3. If we defer to Q3 I'd love to capture this requirement in a ticket before we merge this PR.

**EDIT**: Leaving the above comment as it's still relevant, but less so.

In `install/tests/Makefile` we do provide mechanisms for setting the exact configuration so we should be able to reproduce a given test case if we omit werft. This reduces the challenge of randomizing these values 🎉  , but we'll still need to account for this in the future.

File: install/tests/Makefile
Comment (adrienthebo) [inline install/tests/Makefile:127]:
```suggestion
kots-upgrade:
```

(Tiny nitpick, I think this may have already been addressed in a previous PR)

File: install/tests/Makefile
Comment (adrienthebo) [inline install/tests/Makefile:140]: Note: will this subdomain need to change given the issues we've had with domain level wildcard letsencrypt issuance?

File: install/infra/terraform/gke/main.tf
Comment (nandajavarma) [reply install/infra/terraform/gke/main.tf:160]: This is a good point, ideally we can make this configurable and use a cheaper one for tests. I was basically just translating the [reference architecture](https://www.gitpod.io/docs/self-hosted/latest/reference-architecture/production-ready-single-cluster) here. I was thinking that since our test setup is ephemeral the extra cost we pay here is quite small. May be I am wrong, and it adds up. I will make a note for this for a follow-up

File: install/infra/terraform/k3s/main.tf
Comment (nandajavarma) [reply install/infra/terraform/k3s/main.tf:174]: You are right! I just wanted to avoid calling yet another module from the `main.tf` :see_no_evil:

File: install/tests/Makefile
Comment (nandajavarma) [reply install/tests/Makefile]: We will highly likely stick with the same subdomain, just we will try and re-use the certs issued across multiple sub-domains(max 200 subdomains can be added under a certificate, iirc). In reality, I don't think we will hit a limit even if we create new certs per domain. This week it so happened because I was running too many tests on my own

**Review by adrienthebo — APPROVED** [review]
LGTM!

**Pull request merged**
